internal/pkg/aws/cloudwatch/cloudwatch.go

```diff
@@ -0,0 +1,170 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package cloudwatch provides a client to make API requests to Amazon CloudWatch Service.
+package cloudwatch
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/aws/aws-sdk-go/service/resourcegroups"
+)
+
+const (
+	resourceQueryType      = "TAG_FILTERS_1_0"
+	cloudwatchResourceType = "AWS::CloudWatch::Alarm"
+	compositeAlarmType     = "Composite"
+	metricAlarmType        = "Metric"
+)
+
+type cwClient interface {
+	DescribeAlarms(input *cloudwatch.DescribeAlarmsInput) (*cloudwatch.DescribeAlarmsOutput, error)
+}
+
+type resourceGroupClient interface {
+	SearchResources(input *resourcegroups.SearchResourcesInput) (*resourcegroups.SearchResourcesOutput, error)
+}
+
+// CloudWatch wraps an Amazon CloudWatch client.
+type CloudWatch struct {
+	cwClient
+	resourceGroupClient
```

Comment:
Probably in the future we can pull the resourceGroupClient out - but right now this is fine :D